src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -2811,7 +2812,57 @@ void Scene3D::OnDropped(const QString &_drop, int _mouseX, int _mouseY)
   math::Vector3d pos = renderWindow->ScreenToScene({_mouseX, _mouseY});
 
   msgs::EntityFactory req;
-  req.set_sdf_filename(_drop.toStdString());
+  std::string dropStr = _drop.toStdString();
+  if (QUrl(_drop).isLocalFile())
+  {
+    // mesh to sdf model
+    common::rtrim(dropStr);
+    std::string lowerStr = common::lowercase(dropStr);
+
+    if (!common::EndsWith(lowerStr, ".dae")
+          && !common::EndsWith(lowerStr, ".stl"))
+    {
+      ignwarn << "Only DAE and STL meshes are supported."
```

Comment:
I think we should surface this as an actual dialog or warning.  From a UX perspective, you would drag and drop into the GUI and then nothing would actually happen.  If you didn't think to check the console log, it would be unintuitive.

Reply:
Ahh how I wish we had a proper snackbar https://github.com/ignitionrobotics/ign-gui/issues/44

Reply:
How's this? 7717bac063cf7940e38c6afff8cb440a5887d524

![Peek 2021-07-30 16-40](https://user-images.githubusercontent.com/8602001/127721303-6e9e0985-999f-445c-a61b-bd6d153d5c97.gif)

Reply:
Is there any way to add an okay button to the dialog?  It seems a bit unintuitive.

Reply:
38893b969b0a300f548a68f19e9437e2328d9125 ?
![Peek 2021-08-02 16-22](https://user-images.githubusercontent.com/8602001/127936011-ef747f8c-51e1-4967-aab6-88bd29b4c238.gif)